app/src/test/java/com/example/habitsmasher/test/ListenersTest.java

```diff
@@ -0,0 +1,135 @@
+ */
+public class ListenersTest {
+    private static final String TAG = "Example tag";
+    private static final String MSG = "Example message";
+    private static final String TOAST_MSG = "Toast message";
+    private static final String DAY = "MO";
+    private static final int ID = 1;
+    private static final int POSITION = 0;
+
+    private ClickListenerForCancel _clickListenerForCancel;
+    private ClickListenerForDatePicker _clickListenerForDatePicker;
+    private ClickListenerForDaysOfTheWeek _clickListenerForDaysOfTheWeek;
+    private FailureListener _failureListener;
+    private FailureListenerWithToast _failureListenerWithToast;
+    private SuccessListener _successListener;
+    private SuccessListenerWithToast _successListenerWithToast;
+    private SwipeListener _swipeListener;
+
+    @Before
+    public void setUp() {
+        _clickListenerForCancel = Mockito.spy(new ClickListenerForCancel(mock(Dialog.class), TAG));
+        _clickListenerForDatePicker = Mockito.spy(new ClickListenerForDatePicker
+                (mock(FragmentManager.class), mock(TextView.class)));
+        _clickListenerForDaysOfTheWeek = Mockito.spy(new ClickListenerForDaysOfTheWeek
+                (mock(DaysTracker.class), DAY));
+        _failureListener = Mockito.spy(new FailureListener(TAG, MSG));
+        _failureListenerWithToast = Mockito.spy(new FailureListenerWithToast(mock(Context.class),
+                TAG, MSG, TOAST_MSG));
+        _successListener = Mockito.spy(new SuccessListener(TAG, MSG));
+        _successListenerWithToast = Mockito.spy(new SuccessListenerWithToast(mock(Context.class),
+                TAG, MSG, TOAST_MSG));
+        _swipeListener = Mockito.spy(new SwipeListener(mock(HabitItemAdapter.class),
+                mock(HabitListFragment.class), mock(HabitList.class), mock(User.class)));
+    }
+
+    @Test
+    public void clickListenerForCancel_validInput_expectNoErrors(){
+        View mockView = mock(View.class);
+        _clickListenerForCancel.onClick(mockView);
+        //Here we can verify whether the methods "onClick" executed
+        verify(_clickListenerForCancel, times(1)).onClick(mockView);
+    }
+
+    @Test
+    public void clickListenerForDatePicker_validInput_expectNoErrors(){
+        View mockView = mock(View.class);
+        _clickListenerForDatePicker.onClick(mockView);
+        //Here we can verify whether the methods "onClick" executed
+        verify(_clickListenerForDatePicker, times(1)).onClick(mockView);
+    }
+
+    @Test
+    public void clickListenerForDaysOfTheWeek_validInput_expectNoErrors(){
+        View mockView = mock(View.class);
+        _clickListenerForDaysOfTheWeek.onClick(mockView);
+        //Here we can verify whether the methods "onClick" executed
+        verify(_clickListenerForDaysOfTheWeek, times(1)).onClick(mockView);
+    }
+
+    @Test
+    public void failureListener_validInput_expectNoErrors(){
+        Exception mockException = mock(Exception.class);
```

Comment:
Can we be more specific about the **type** of exception we expeect here?

Reply:
Any updates on this? 

I see we are throwing a `NullPointerException`, so we can probably be more specific on the **type** of Exception we expect. If literally **anything** goes wrong, this test will still pass, because we are expecting **any Exception**. The same thing applies to any other test that does this.

- [ ] change Exception to NullPointerException